### PR TITLE
osd/PG: reset PG on osd down->up; normalize query processing

### PR DESCRIPTION
**Pull request by liewegas**
http://tracker.ceph.com/issues/24373

### CODE REVIEW

**Comment by liewegas** [comment]
@skouya how about this?

File: src/osd/PG.cc [hunk 1]
```diff
@@ -5765,9 +5765,12 @@ bool PG::should_restart_peering(
     dout(20) << "new interval newup " << newup
 	     << " newacting " << newacting << dendl;
     return true;
-  } else {
-    return false;
   }
+  if (!lastmap->is_up(osd->whoami) && osdmap->is_up(osd->whoami)) {
+    dout(10) << __func__ << " osd transitioned from down -> up" << dendl;
+    //return true;
```
Comment (neha-ojha) [inline src/osd/PG.cc]: you mean to return true here, right?

**Comment by liewegas** [comment]
yep!  /facepalm

**Comment by skouya** [comment]
Hi Sage,
Thank you for your quick fix!
It sounds good to me.

I confirmed your fix resolves my issue:
http://tracker.ceph.com/issues/24373

Thanks a lot.
Kouya

**Pull request merged**